.github/workflows/test.yml

```diff
@@ -104,60 +92,63 @@ jobs:
         run: |
           echo "Full test in pr : ${{ steps.check_request.outputs.full_tests_in_pr }}"
-          if ${{steps.check_if_release_pr.outputs.charts_release_branch == 'true' || steps.check_request.outputs.full_tests_in_pr == 'true' }} ; then
-            echo "Release PR from dev to charts, oer PR with new full test, so running full tests"
-            ve1/bin/behave tests/functional/behave_features/ --tags=full --logging-level=WARNING --no-capture --no-color
-          else
-            echo "Not a release PR from dev to charts, so running only smoke tests"
-            ve1/bin/behave tests/functional/behave_features/ --tags=smoke --logging-level=WARNING --no-capture --no-color
+          echo "Is charts release branch : ${{ steps.check_request.outputs.full_tests_in_pr }}"
+          if ${{ steps.check_if_release_pr.outputs.charts_release_branch == 'true' || steps.check_request.outputs.full_tests_in_pr == 'true' }} ; then
+            echo "Release PR from dev to charts, or PR with new full test, so running full tests"
+            echo "test-tags=full" | tee -a $GITHUB_OUTPUT
+            exit 0
           fi
 
-      - name: (Manual) Test CI Workflow
-        if: |
-          github.event_name == 'workflow_dispatch' && steps.check_request.outputs.run-tests == 'true'
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DRY_RUN: ${{ github.event.inputs.dry-run }}
-          VENDOR_TYPE: ${{ github.event.inputs.vendor-type }}
-          NOTIFY_ID: ${{ github.event.inputs.notify-id }}
-          SOFTWARE_NAME: ${{ github.event.inputs.software-name }}
-          SOFTWARE_VERSION: ${{ github.event.inputs.software-version }}
-          BOT_NAME: ${{ secrets.BOT_NAME }}
-          BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
-          PR_BODY: "Triggerd by ${{ github.event.sender.html_url }} from ${{ github.event.repository.html_url }} on `${{ github.event.ref }}`."
-        run: |
-          echo "[INFO] Dry run '${{ env.DRY_RUN }}'"
-          echo "[INFO] Vendor type '${{ env.VENDOR_TYPE }}'"
-          echo "[INFO] Notify ID '${{ env.NOTIFY_ID }}'"
-          echo "[INFO] Software Name '${{ env.SOFTWARE_NAME }}'"
-          echo "[INFO] Software Version '${{ env.SOFTWARE_VERSION }}'"
-          ve1/bin/behave tests/functional/behave_features/ --tags=version-change --logging-level=WARNING --no-capture --no-color
-
-      - name: Approve PR
-        id: approve_pr
-        if: ${{ steps.check_if_release_pr.outputs.charts_release_branch == 'true' }}
-        uses: hmarr/auto-approve-action@v4
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          echo "Not a release PR from dev to charts, so running only smoke tests"
+          echo "test-tags=smoke" | tee -a $GITHUB_OUTPUT
+  
+  run-tests:
+    # No further pull request author checking done here because
+    # check-contributor gates the needed jobs.
+    name: Run Tests
+    needs:
+    - determine-workflow-conditions
```

Comment:
I know it doesn't really change anything, but should it also need `check-contributor` to make the running order more explicit ?

I'm not sure what the best practice is (specifying all jobs that have to run before vs specifying only the previous job), but this is usually what I've seen in our repo.

Reply:
Good catch. This is similar to the discussion we had [here](https://github.com/openshift-helm-charts/development/pull/392/files/6a88a5d2998adb24c637381da8cb7ab6a15cb3ba#r1806729666) and I'm open to all options here.

GitHub "puts these in a line" from a dependency perspective, because *determine-workflow-conditions* needs *check-contributor*. Beyond documentation clarity: E.g. "this job should be contributor-gated, and is as written", the only other reason to add something to the *needs* list from GitHub's perspect is to use its outputs as inputs to the dependent job.

From a documentation perspective, I've added two lines of comment for something that could have been one line of comment + an entry in the *needs* list. With that in mind, I'm open to doing this either way. It wouldn't change the execution as far as I can tell.